Automattic/vip-go-mu-plugins
# Search Dev Tools: Differentiate between search and indexing rate limits

## Description
We currently don't differentiate what type of rate limiting is happening in place in Search Dev Tools.
This PR differentiates between the two types of rate limiting happening in Search Dev Tools

**Scenarios:**

Scenario 0) When no rate limiting happening:

<img width="1189" alt="Screenshot 2023-03-15 at 2 51 05 PM" src="https://user-images.githubusercontent.com/16962021/225439603-2f194163-72cc-46c7-87d3-0027342b97bc.png">

Scenario 1) When rate limiting happening on indexing, but not search
<img width="1266" alt="Screenshot 2023-03-15 at 2 53 16 PM" src="https://user-images.githubusercontent.com/16962021/225439786-2a773cc8-9011-4840-b3cd-789f717c340d.png">

Scenario 2) When rate limiting happening on search, but not indexing:
<img width="1285" alt="Screenshot 2023-03-15 at 2 59 43 PM" src="https://user-images.githubusercontent.com/16962021/225441173-456fdbdf-e98a-4f2f-80b3-b16443915788.png">

Scenario 3) When rate limiting happening on both types:
<img width="1289" alt="Screenshot 2023-03-15 at 2 57 08 PM" src="https://user-images.githubusercontent.com/16962021/225440790-5be87e76-5a8e-442d-b0f3-8ec8f6a052d6.png">


## Changelog Description

### Plugin Updated: Search Dev Tools

If being rate limited, differentiate between two different types: "search" or "indexing"

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test
0) Go to Search Dev Tools and see scenario 0) From Description
1) Fake an indexing rate limiting in `wp shell`:
```
$search = \Automattic\VIP\Search\Search::instance();
$search->queue->turn_on_index_ratelimiting();
$search->queue->handle_index_limiting_start_timestamp();
```
2) Go to Search Dev Tools and see scenario 1) from Description
3) Fake a search rate limiting in `wp shell`:
```
wp_cache_set( 'query_count', 100000, 'vip_search' );
```
4) Go to Search Dev Tools and see Scenario 3) from Description
5) Wait 5 minutes to wait for indexing rate limit to expire and visit Search Dev Tools to see Scenario 2) From Description
